### PR TITLE
Added new scanner for scanning only workloads of cluster

### PR DESCRIPTION
**Pull request by snootan**
## Description
This change will create new scanner for scanning only workloads of any cluster. 
It will not inspect workloads for checking vulnerabilities.

### CODE REVIEW

File: src/pkg/inspectors/workloadscanner/pkg/controller.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+	ExportImageReports(report, policy)
+
+	return nil
+}
+
+func ExportImageReports(report itypes.AssessmentReport, pl *v1alpha1.InspectionPolicy) {
+	if bytes, err := json.Marshal(report); err != nil {
+		// Marshal failure should be fatal because it is unforgivable
+		log.Fatal(err, "failed to marshal the insight struct")
```
Comment (JingChen23) [inline src/pkg/inspectors/workloadscanner/pkg/controller.go:171]: Well, we need to change the log.

File: src/pkg/inspectors/workloadscanner/pkg/controller.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/pkg/errors"
+	"github.com/vmware-tanzu/cloud-native-security-inspector/src/api/v1alpha1"
+	wl "github.com/vmware-tanzu/cloud-native-security-inspector/src/lib/assets/workload"
+	"github.com/vmware-tanzu/cloud-native-security-inspector/src/lib/log"
+	exporter_inputs "github.com/vmware-tanzu/cloud-native-security-inspector/src/pkg/exporter/inputs"
+	itypes "github.com/vmware-tanzu/cloud-native-security-inspector/src/pkg/inspectors/imagescanner/types"
```
Comment (JingChen23) [inline src/pkg/inspectors/workloadscanner/pkg/controller.go:13]: You assessmentReport is from this pkg? Then why are your defining a new one in src/pkg/inspectors/workloadscanner/types/report.go？

Comment (JingChen23) [reply src/pkg/inspectors/workloadscanner/pkg/controller.go:13]: itypes.AssessmentReport on line 168

**Comment by snootan** [comment]
> Apart from the above comment, the rest of the changes look good. Can you share sample assessment report getting generated?

@bgubbala, exporter has not implemented yet. Once it will get implemented then we could see that report.
But, we could see the report which is sending from workload scanner.
Attaching screenshot:
Screenshot 2023-03-10 at 11 02 54 AM

**Review by JingChen23 — APPROVED** [review]

File: src/config/samples/policy.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ spec:
     image: projects.registry.vmware.com/cnsi/inspector:0.3
     kubebenchImage: projects.registry.vmware.com/cnsi/kubebench:0.3
     riskImage: projects.registry.vmware.com/cnsi/risk:0.3
-    governorImage:
+    workloadScanner: projects.registry.vmware.com/cnsi/workloadscanner:0.3
```
Comment (bgubbala) [inline src/config/samples/policy.yaml:18]: I don't see inspectionpolicy_types.go changes here, CRD bases have to be generated from types.

https://github.com/vmware-tanzu/cloud-native-security-inspector/blob/main/src/api/v1alpha1/inspectionpolicy_types.go

File: src/config/crd/bases/goharbor.goharbor.io_inspectionpolicies.yaml [hunk 4]
```diff
@@ -298,6 +298,9 @@ spec:
                   riskImage:
                     description: Image of the risk.
                     type: string
+                  workloadScanner:
```
Comment (bgubbala) [inline src/config/crd/bases/goharbor.goharbor.io_inspectionpolicies.yaml:301]: I don't see `inspectionpolicy_types.go `changes here, CRD bases have to be generated from types.

https://github.com/vmware-tanzu/cloud-native-security-inspector/blob/main/src/api/v1alpha1/inspectionpolicy_types.go

**Review by bgubbala — CHANGES_REQUESTED** [review]

**Comment by JingChen23** [comment]
> Can you confirm?
@bgubbala 
You can send your report to OpenSearch right now, the payload is a json string. Such as:
```
"{\"docId\":\"container1-assessment-report\",\"NodeName\":\"node1\",\"ClusterName\":\"cluster1\",\"payload\":{\"checkitem\":\"1\"}}"
```
If you want to extend the export to send one copy to governor,  you need to extend the Exporter.
After that, in the policy, you can only enable governor, then Exporter will not send a copy to OpenSearch, and will only send to Governor.

**Review by JingChen23 — APPROVED** [review]
LGTM

**Review by bgubbala — APPROVED** [review]

**Pull request merged**